File: Code/javascript/LZ77-Kompression/src/Util/Dictionary.js
class Dictionary {

    constructor(){
        this.dictionary = new Array();
    }

    createEntry(offset, length, nextSymbol){
        const entry = new Entry(offset, length, nextSymbol);
        this.dictionary.add(entry);
    }

    addEntry(entry){
        this.dictionary.add(entry);
    }

    getDictionary() {
        return this.dictionary;
    }
}

File: Code/Java/LZ77-Compression/src/main/java/org/example/Util/Dictionary.java
package org.example.Util;

import java.util.ArrayList;

public class Dictionary {
    private final ArrayList<Entry>  dictionary;

    public Dictionary(){
        dictionary = new ArrayList<Entry>();
    }

    public void createEntry(int offset, int length, char nextSymbol){
        Entry entry = new Entry(offset,length,nextSymbol);
        dictionary.add(entry);
    }

    public void addEntry(Entry entry){
        dictionary.add(entry);
    }

    public ArrayList<Entry> getDictionary() {
        return dictionary;
    }
}
